indy-utils/src/main/kotlin/com/luxoft/blockchainlab/hyperledger/indy/helpers/ConfigHelper.kt

```diff
@@ -0,0 +1,83 @@
+package com.luxoft.blockchainlab.hyperledger.indy.helpers
+
+import com.natpryce.konfig.*
+import java.io.File
+
+/**
+ * Helps with config retrieving
+ */
+object ConfigHelper : IndyConfiguration {
+    private val CFG_PATH = "indyconfig"
+    private val CFG_NAME = "indy.properties"
+
+    /**
+     * Returns [indyuser] [Configuration]
+     *
+     * If looks in test configuration, then it looks in file `indyconfig/indy.properties`,
+     * then it looks in environment variables.
+     * When using environment variables to pass config make sure you've named your vars correctly:
+     * indy.genesisFile will become INDY_GENESISFILE and so on.
+     *
+     * @return: [Configuration] [indyuser]
+     * @throws: [KotlinNullPointerException] - if no configuration at all was found
+     */
+    fun getConfig(path: String = CFG_PATH, name: String = CFG_NAME): Configuration {
```

Comment:
You will re-read file every time when ConfigHelper functions are called. This is a constant file and it's used once on start.

Reply:
Actually, you're not right. Look at CordentityE2E test. ConfigHelper is mocked there and it is Kotlin-object - so it is loaded in memory for only *once*. We have like 4 nodes in this test. What will happen?

Reply:
It should either not be Kotlin-object or re-read file every time, to make local tests work.

Reply:
No, test was specially adjusted to support Object in Kotlin. If you remove the test (make it old manner) it will still work and you will see that file is re-read every time. This is a reason why IndyMock was created. The idea is to have just one indy.property per instance and don't read file many times.

Reply:
Why not make it config val and init lazy way??

Reply:
Yes, this it true, it should be val. There is no need to have getConfig as a function at all. Variable should be initialized in advance and file won't be read multiple times then.

Reply:
If u say so.